Cabal/Distribution/Simple/GHC/IPIConvert.hs

```diff
@@ -9,31 +9,31 @@
 --
 -- Helper functions for 'Distribution.Simple.GHC.IPI642'.
 module Distribution.Simple.GHC.IPIConvert (
-    PackageIdentifier, convertPackageId,
+    MungedId, convertMungedId,
     License, convertLicense,
     convertModuleName
   ) where
 
 import Prelude ()
 import Distribution.Compat.Prelude
 
-import qualified Distribution.Types.PackageId as Current
-import qualified Distribution.Types.PackageName as Current
+import qualified Distribution.Types.MungedId as Current
+import qualified Distribution.Types.MungedName as Current
 import qualified Distribution.License as Current
 
 import Distribution.Version
 import Distribution.ModuleName
 import Distribution.Text
 
-data PackageIdentifier = PackageIdentifier {
-    pkgName    :: String,
-    pkgVersion :: Version
+data MungedId = MungedId {
```

Comment:
Not only does the constructor have to be, but each record selector has to be the same.

```
ezyang@sabre:~$ runghc T.hs
MkT {f = 2}
ezyang@sabre:~$ cat T.hs
data T = MkT {f :: Int }
    deriving (Show)

main = print (MkT 2)
```